### PR TITLE
feat(@clayui/table): Adds ClayTableWithEditableRow

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

File: packages/clay-table/src/Row.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import classNames from 'classnames';
 import React from 'react';
 import {TDelimiter} from './types';
 
-interface IRowProps extends React.HTMLAttributes<HTMLTableRowElement> {
+export interface IRowProps extends React.HTMLAttributes<HTMLTableRowElement> {
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply packages/clay-table/src/Row.tsx]: Great! :)

File: packages/clay-table/src/ClayTableWithEditableRow.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+				</Cell>
+			)}
+			{cells.map(({cellProps, editable = false, title}, index) => (
+				<Cell expanded key={index} {...cellProps}>
+					{editable && editing ? (
+						<ClayInput
+							className="form-control-sm"
+							defaultValue={title}
+							onChange={event => {
+								cells[index].title = event.target.value;
```
[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-table/src/ClayTableWithEditableRow.tsx]: This is really a bad practice, these values ​​may be lost on the next render. We must add a `onInputChange` callback or issue a `onCellChange`.

File: packages/clay-table/src/EditableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+	/**
+	 * Callback function for when the selectable checkbox state changes.
+	 */
+	onCheckboxChange?: (isChecked: boolean) => void;
+
+	/**
+	 * Callback function for when the `Save` button is clicked.
+	 */
+	onRowUpdated?: (newEntries: Array<IEditableCell>) => void;
```
Comment (matuzalemsteles) [inline packages/clay-table/src/EditableRow.tsx]: Maybe `onRowChange`? This creates better consistency with the other API's.

File: packages/clay-table/src/Row.tsx [hunk 4]
```diff
@@ -7,7 +7,6 @@
 import classNames from 'classnames';
 import React from 'react';
 import {TDelimiter} from './types';
-
```
Comment (matuzalemsteles) [inline packages/clay-table/src/Row.tsx]: I think we should keep this line blank.

File: packages/clay-table/src/ClayTableWithEditableRow.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+interface IEditableCell {
+	cellProps?: Object;
+	editable?: boolean;
+	title: string;
+}
+
+interface IEditableRowProps extends IRowProps {
+	/**
+	 * Defines the Cell.
```
[the remark this answers is not in the document]
Comment (diegonvs) [reply packages/clay-table/src/ClayTableWithEditableRow.tsx]: Updated!

File: packages/clay-table/src/ClayTableWithEditableRow.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (diegonvs) [reply packages/clay-table/src/ClayTableWithEditableRow.tsx]: Solved creating a copy of `cells` for avoiding mutating props.

File: packages/clay-table/src/EditableRow.tsx (hunk 3 above)
Comment (diegonvs) [reply packages/clay-table/src/EditableRow.tsx]: Makes sense Matu!

**Comment by bryceosterhaus** [comment]
Going to close this for now and will re-open if it becomes relevant again for us when we add new table components

**Pull request closed without merging**